client/web/web.go

```diff
@@ -100,48 +100,21 @@ func init() {
 	tmpls = template.Must(template.New("").ParseFS(embeddedFS, "*"))
 }
 
-// authorize returns the name of the user accessing the web UI after verifying
-// whether the user has access to the web UI. The function will write the
-// error to the provided http.ResponseWriter.
-// Note: This is different from a tailscale user, and is typically the local
-// user on the node.
-func authorize(w http.ResponseWriter, r *http.Request) (string, error) {
+// ServeHTTP processes all requests for the Tailscale web client.
+func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	switch distro.Get() {
 	case distro.Synology:
-		user, err := synoAuthn()
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusUnauthorized)
-			return "", err
-		}
-		if err := authorizeSynology(user); err != nil {
-			http.Error(w, err.Error(), http.StatusForbidden)
-			return "", err
-		}
-		return user, nil
+		s.serveSynology(w, r)
+		return
 	case distro.QNAP:
-		user, resp, err := qnapAuthn(r)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusUnauthorized)
-			return "", err
-		}
-		if resp.IsAdmin == 0 {
-			http.Error(w, err.Error(), http.StatusForbidden)
-			return "", err
-		}
-		return user, nil
-	}
-	return "", nil
-}
-
-func authRedirect(w http.ResponseWriter, r *http.Request) bool {
-	if distro.Get() == distro.Synology {
-		return synoTokenRedirect(w, r)
+		s.serveQNAP(w, r)
+		return
+	default:
+		s.serve(w, r)
```

Comment:
Since the above two eventually land at `s.serve`, maybe we refactor those to be `authorizeQNAP` and `authorizeSynology` and then remove the default here and have everything fall to `s.serve`?

```
var err error
switch distro.Get() {
	case distro.Synology:
		err = s.authorizeSynology(w, r)
        case distro.QNAP:
                err = s.authorizeQnap(w, r)
}
if err != nil {
        http.Error(w, err.Error(), http.StatusForbidden)
	return
}
s.serve(w, r)
```

That way it's very obvious that they all use `serve`. At a quick glance without knowing that, it might seem like more logic differs than just the authorization. Might also be worth a tiny comment about why we only have auth cases for Synology and QNAP.

Reply:
done